### PR TITLE
MAD Quests

### PR DESCRIPTION
**Pull request by LegitDongo**
# Description
Quests!
Merry Christmas guys, this gives MAD compatibility for quests!

Check out the included documentation on the new DTS and filters. These should be pretty inclusive to what you guys want, if there's more that's needed, let me know and I'll get it added.

Translations are probably a little rough (even though most of them I got from the game), so please comment where they are wrong.

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (would cause existing functionality to change)

## Motivation and Context
Because people want them quests

## How Has This Been Tested?
Very lightly, windows 10, webhook tester

## Wiki Update
- [x] This change requires an update to the Wiki.
- [ ] This change does not require an update to the Wiki.

Wiki update already done

### CODE REVIEW

File: PokeAlarm/Events/QuestEvent.py [hunk 1]
```diff
@@ -19,10 +21,9 @@ def __init__(self, data):
         # Identification
         self.stop_id = data['pokestop_id']
         self.stop_name = check_for_none(
-            str, data.get('pokestop_name') or data.get('name'),
-            Unknown.REGULAR)
+            str, data.get('pokestop_name', data.get('name')), Unknown.REGULAR)
         self.stop_image = check_for_none(
-            str, data.get('pokestop_url') or data.get('url'), Unknown.REGULAR)
+            str, data.get('pokestop_url', data.get('url')), Unknown.REGULAR)
```
Comment (kvangent) [inline PokeAlarm/Events/QuestEvent.py:26]: This looks wrong - what are you trying to do?

File: PokeAlarm/Events/QuestEvent.py [hunk 2]
```diff
@@ -38,28 +39,43 @@ def __init__(self, data):
...
-        self.quest = data['quest']
-        self.reward = data['reward']
-        # To ensure Windows compatibility, epoch calculation is required
-        self.expire_time = datetime.utcfromtimestamp(
-            data.get(
-                'expire_time',
-                (datetime.combine(date.today(), time(23, 59)) -
-                 datetime(1970, 1, 1)).total_seconds()))
-        self.reward_type = check_for_none(int, data.get('type'), 0)
+        self.quest_type_raw = data['quest_type']
```
Comment (kvangent) [inline PokeAlarm/Events/QuestEvent.py:42]: Are all these fields requires required? right now it'll throw an error for any that don't exist

File: PokeAlarm/Utilities/QuestUtils.py [hunk 3]
```diff
@@ -27,3 +27,56 @@ def get_reward_type(reward_type):
     except ValueError:
         raise ValueError("Unable to interpret `{}` as a valid "
                          " quest reward type or id.".format(reward_type))
+
+
+def reward_string(quest, locale):
+    if quest.reward_type_id == 7:
```
Comment (kvangent) [inline PokeAlarm/Utilities/QuestUtils.py]: Can you add some comments or clarification to this function? Not quite sure what the significance of these ids are, or why they mean different things.

File: docs/configuration/alarms/discord.rst [hunk 4]
```diff
@@ -193,6 +193,15 @@ Example: Alarm Configuration Using Optional Parameters
           "title":"This Pokestop has been invaded by Team Rocket!",
           "url":"<gmaps>",
           "body":"Invasion will expire at <24h_time> (<time_left>)"
+      },
+      "quests":{
+          "webhook_url":"DISCORD_WEBHOOK_URL_FOR_INVASION_CHANNEL",
+          "username":"Quest",
+          "icon_url*":"<YOUR CUSTOM URL HERE>/<quest_image>.png",
```
Comment (kvangent) [inline docs/configuration/alarms/discord.rst]: We should be consistent with values we want the user to replace - also consider avoiding `<>` as that currently signals a DTS. 
```suggestion
          "icon_url*":"<YOUR_CUSTOM_URL_HERE>/<quest_image>.png",
```

**Review by kvangent — CHANGES_REQUESTED** [review]

File: PokeAlarm/Events/QuestEvent.py (hunk 1 above)
Comment (LegitDongo) [reply PokeAlarm/Events/QuestEvent.py:26]: see #691 - sometimes `or` evaluates things as falsey, this seems like the best remedy

**Comment by LegitDongo** [comment]
For those having issues applying this patch, try this:
`git apply --reject --whitespace=fix 706.patch`
this will fix whitespace issues and make new files for rejected changes

en.json will need things replaced based off rejected changes
`quest_types` -> `quest_reward_types`

File: PokeAlarm/Events/QuestEvent.py (hunk 2 above)
Comment (kvangent) [reply PokeAlarm/Events/QuestEvent.py:42]: These should probably be .gets for backwards compatibility

**Review by kvangent — CHANGES_REQUESTED** [review]

**Review by kvangent — APPROVED** [review]

**Pull request merged**